examples/tasks/async_compute.rs

```diff
@@ -0,0 +1,122 @@
+use bevy::prelude::*;
+use bevy::tasks::{AsyncComputeTaskPool, Task};
+use std::time::{Instant, Duration};
+use rand::Rng;
+use std::option::Option::Some;
+use futures_lite::future;
+
+/// This example shows how to use the ecs system and the AsyncComputeTaskPool
+/// to spawn, poll, and complete tasks across systems and system ticks.
+
+// Number of cubes to spawn across the x, y, and z axis
+const NUM_CUBES: i32 = 6;
+
+// Used to tag our new entity spawned with tasks
+struct Marker;

```

Comment:
Wait we don't need this at all. We can just query for the Task component!

Reply:
Wouldn't most use cases be using a tagged entity? At least in mine it does. If you have different entity archetypes that happen to have the same Task<T> component you wouldn't want to return all of them.

Reply:
For delayed asset loading for example I'd want to grab all the tasks of that type that are ready.

Reply:
> For delayed asset loading for example I'd want to grab all the tasks of that type that are ready.

The subsequent work to be done on the result may vary based on the archetype it belongs to., right?

Reply:
Hmm, often. 

I suppose I feel that splitting that sort of work into different queries is a fairly introductory concept that isn't specific to the pattern this example is teaching.

In the end, my view is that having what is effectively dead code with the `Marker` in this example obscures what's actually going on, and isn't motivated enough to help the beginners who would actually benefit from seeing that pattern in action.

Reply:
Changed in e065dff